WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,232 @@
+    bigStep = 1,
+    default = 30,
+  },
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+  end,
+  ["OrientationChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+    self:SetTickThickness(self.tick_thickness, true)
+  end,
+  ["OnSizeChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+  end,
+  SetVisible = function(self, visible)
+    if self.tick_visible ~= visible then
+      self.tick_visible = visible
+      if visible then
+        self:Show()
+      else
+        self:Hide()
+      end
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.tick_color[1], self.tick_color[2], self.tick_color[3], self.tick_color[4] = r, g, b, a or 1
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacementMode = function(self, placement_mode)
+    if self.placement_mode ~= placement_mode then
+      self.placement_mode = placement_mode
+      self:SetTickPlacement(self.tick_placement)
+    end
+  end,
+  SetTickPlacement = function(self, placement, forced)
+    if forced or self.tick_placement ~= placement then
+      self.tick_placement = placement
+      self.parentOrientation = self.parent.effectiveOrientation
+      self.parentInverse = self.parent.inverseDirection
+      if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+        self.parentTrueHeight, self.parentTrueWidth = self.parent.bar:GetRealSize()
+      else
+        self.parentTrueWidth, self.parentTrueHeight = self.parent.bar:GetRealSize()
+      end
+      local offset, offsetx, offsety = self.tick_placement, 0, 0
+      local width = self.parentTrueWidth
+      if self.tick_placement_mode == "ABSOLUTE" then
+        local pixels = width / (self.parent.state.duration or self.parent.state.total or 1)
+        offset = math.max(math.min((placement * pixels), width), 0)
+      elseif self.tick_placement_mode == "RELATIVE" then
+        offset = (placement / 100) * width
+      end
+
+      if self.parentInverse ~= ((self.parentOrientation == "HORIZONTAL_INVERSE") or (self.parentOrientation == "VERTICAL")) then
```

Comment:
I wonder whether this cleaner as:
local inverse = self.parentInverse
if((self.parentOrientation == "HORIZONTAL_INVERSE") or (self.parentOrientation == "VERTICAL")) then
  inverse = not inverse
end

if inverse then
  offset = -offset
end